lib/_http_client.js

```diff
@@ -618,8 +618,11 @@ function responseOnEnd() {
   const req = this.req;
 
   req._ended = true;
-  if (!req.shouldKeepAlive || req.finished)
+  if (!req.shouldKeepAlive || req.finished) {
+    req.emit('close');
```

Comment:
Needs test?

Reply:
Definitely needs a test. Will review again once that's added

Reply:
Resolved in https://github.com/nodejs/node/commit/de246fc0867077ac672467d33a59892ec3dc1a29